src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -218,6 +218,44 @@ public class SqlMutationEngine : IMutationEngine
             return result;
         }
 
+        /// <summary>
+        /// Helper method to determine:
+        /// 1. Whether a mutation is a mutate one or mutate many operation (eg. createBook/createBooks)
+        /// 2. Name of the top-level entity backing the mutation.
+        /// </summary>
+        /// <param name="context">GraphQL request context.</param>
+        /// <returns>a tuple of the above mentioned metadata.</returns>
+        private static Tuple<bool, string> GetMutationCategoryAndEntityName(IMiddlewareContext context)
+        {
+            IOutputType outputType = context.Selection.Field.Type;
+            string entityName = string.Empty;
+            ObjectType _underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
+            bool isPointMutation;
+            if (GraphQLUtils.TryExtractGraphQLFieldModelName(_underlyingFieldType.Directives, out string? modelName))
+            {
+                isPointMutation = true;
+                entityName = modelName;
+            }
+            else
+            {
+                // Absence of model directive on the mutation indicates that we are dealing with a `mutate many`
+                // mutation like createBooks.
+                isPointMutation = false;
+
+                // For a mutation like createBooks which inserts multiple records into the Book entity, the underlying field type is a paginated response
+                // type like 'BookConnection'. To determine the underlying entity name, we have to look at the type of the `items` field which stores a list of items of
+                // the underlying entity's type - here, Book type.
+                IOutputType entityOutputType = _underlyingFieldType.Fields[MutationBuilder.ARRAY_INPUT_ARGUMENT_NAME].Type;
+                ObjectType _underlyingEntityFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(entityOutputType);
+                if (GraphQLUtils.TryExtractGraphQLFieldModelName(_underlyingEntityFieldType.Directives, out modelName))
```

Comment:
Would we benefit from an else statement here to throw an exception in case any scenarios exist where this condition doesn't hold true?

Reply:
I don't think this condition would ever not be true.